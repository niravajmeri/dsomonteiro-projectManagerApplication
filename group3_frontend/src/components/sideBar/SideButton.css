
.Item {
    font-size: 15px;
    margin-top: 15px;
}

.Item .icon {
    margin-top: -30px;
    margin-right: 20px;
    font-size: 10px;
    color: rgb(195, 195, 195);
}



.Item > a {

    white-space: nowrap;
    color: rgb(221, 221, 221);
    text-decoration: none;
}

.Item a:hover {
    text-decoration: none;
    font-weight: bold;
}

.Item > a.active,
.Item > a.active:focus {
    color: rgb(185, 184, 184);
    text-decoration: none;
}


.ItemButton {
    padding-right: 250px;
    padding-top: 35px;
    padding-left: 15px;
    background: none;
    font-size: 15px;
    box-shadow: 0px 0px 0px transparent;
    border: 0px solid transparent;
    text-shadow: 0px 0px 0px transparent;
    outline: none;
    padding-bottom: 15px;
    border-bottom: 1px solid rgba(150, 150, 150, 0.281);
}

.ItemButton:hover {
    background-color: rgba(27, 43, 56, 0.479);
    box-shadow: 0px 0px 0px transparent;
    border: 0px solid transparent;
    text-shadow: 0px 0px 0px transparent;
    font-weight: bold;
    border-bottom: 1px solid rgba(150, 150, 150, 0.281);

}









