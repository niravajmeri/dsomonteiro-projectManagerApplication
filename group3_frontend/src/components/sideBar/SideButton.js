import React from 'react';
import './SideButton.css';
import { NavLink } from 'react-router-dom';

class SideButton extends React.Component {


    render() {
        let icon = (
            <div className="icon" align="right">
                <span className="glyphicon glyphicon-chevron-right" />
            </div>
        );


        return (
                <div className="Item">
                    <NavLink to={this.props.to}>
                        <button className="ItemButton" >{this.props.text} </button>
                    </NavLink>
                    {icon}
                </div>
        );

    }
}

export default SideButton;
