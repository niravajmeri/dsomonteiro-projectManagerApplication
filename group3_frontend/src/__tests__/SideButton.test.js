import React from 'react';
import { shallow } from 'enzyme';
import SideButton from '../components/sideBar/SideButton';

describe('SideButton', () => {
    it('should render correctly', () => {
        const component = shallow(
            <SideButton to="/mytest" text="Button text" />
        );
        expect(component).toMatchSnapshot();
    });
});
