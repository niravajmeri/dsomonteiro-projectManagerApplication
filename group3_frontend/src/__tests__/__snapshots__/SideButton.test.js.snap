// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideButton should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideButton
    text="Button text"
    to="/mytest"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/mytest"
>
          <button
                    className="ItemButton"
          >
                    Button text
                     
          </button>
</NavLink>,
        <div
          align="right"
          className="icon"
>
          <span
                    className="glyphicon glyphicon-chevron-right"
          />
</div>,
      ],
      "className": "Item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="ItemButton"
>
            Button text
             
</button>,
          "to": "/mytest",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Button text",
              " ",
            ],
            "className": "ItemButton",
          },
          "ref": null,
          "rendered": Array [
            "Button text",
            " ",
          ],
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "align": "right",
          "children": <span
            className="glyphicon glyphicon-chevron-right"
/>,
          "className": "icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "glyphicon glyphicon-chevron-right",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/mytest"
>
            <button
                        className="ItemButton"
            >
                        Button text
                         
            </button>
</NavLink>,
          <div
            align="right"
            className="icon"
>
            <span
                        className="glyphicon glyphicon-chevron-right"
            />
</div>,
        ],
        "className": "Item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="ItemButton"
>
              Button text
               
</button>,
            "to": "/mytest",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Button text",
                " ",
              ],
              "className": "ItemButton",
            },
            "ref": null,
            "rendered": Array [
              "Button text",
              " ",
            ],
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "align": "right",
            "children": <span
              className="glyphicon glyphicon-chevron-right"
/>,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "glyphicon glyphicon-chevron-right",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
